Stat Build Updates

CrumpsBr0ther opened the pull request:
Updates to hero build choices and decouples the relationship between hero build and name.

Commit:
Stat Build Updates

Updates to hero build choices and decouples the relationship between hero build and name.

Commit:
Web Interface

Adding to the web interface. Still not 100% though.

Commit:
The Xarnax42 Patch

Wanna make sure a zero length name takes exactly as long to print in the dialog box as a name of any other length.

Commit:
ALL_BUILDS

Well there's the problem. Little tidbit of code to sort out the tri-stated flags was wrecking the ALL_BUILDS flag. Since nothing is currently in slot #12 except for my stats drop down I just chopped the loop to stop a smidge earlier. So now this whole branch is working exactly like it's supposed to work.

mcgrew commented:
I saw some of the screenshots for this. It's not a terrible idea but I think a change like this needs community feedback before it's implemented.

Commit:
Merge remote-tracking branch 'upstream/master' into big_stat_choices

Commit:
NO_BUILDS user feedback

Made the minimum str 6 and mp 8 (with spells) for the NO_BUILDS options. These values were always available to the runner using specific builds to boost base stats at the start of the game.
